web/embedding.py

```diff
@@ -244,34 +308,48 @@ def _from_word2vec_binary(fname):
     def _from_word2vec_text(fname):
         with _open(fname, 'r') as fin:
             words = []
+
             header = fin.readline()
             ignored = 0
             vocab_size, layer1_size = list(map(int, header.split()))  # throws for invalid file format
             vectors = np.zeros(shape=(vocab_size, layer1_size), dtype=np.float32)
             for line_no, line in enumerate(fin):
                 try:
-                    parts = text_type(line, encoding="utf-8").strip().split()
+                    parts = text_type(line, encoding="utf-8").split(' ')
+                    w = parts[0]
+                    parts = list(map(lambda x: x.strip(), parts[1:]))
+                    parts.insert(0, w)
+
                 except TypeError as e:
-                    parts = line.strip().split()
+                    parts = line.split(' ')
+                    w = parts[0]
+                    parts = list(map(lambda x: x.strip(), parts[1:]))
+                    parts.insert(0, w)
+
                 except Exception as e:
-                    logger.warning("We ignored line number {} because of erros in parsing"
+                    logger.warning("We ignored line number {} because of errors in parsing"
                                    "\n{}".format(line_no, e))
                     continue
+
                 # We differ from Gensim implementation.
                 # Our assumption that a difference of one happens because of having a
                 # space in the word.
                 if len(parts) == layer1_size + 1:
                     word, vectors[line_no - ignored] = parts[0], list(map(np.float32, parts[1:]))
-                elif len(parts) == layer1_size + 2:
+                elif len(parts) == layer1_size + 2 and parts[-1]:
```

Comment:
Add comment explaining why this condition (it is quite cryptic)

Reply:
Done